File: Restore.API/Responses/ErrorResponse.cs
using System.Net;

namespace Restore.API.Responses;

public record ErrorResponse<T>(T Error, bool Success = false);


public record ExceptionResponse(string Message, HttpStatusCode StatusCode, string CorrelationId);

public record ExceptionResponseDev(
    string Message,
    HttpStatusCode StatusCode,
    string? Details,
    string? StackTrace,
    string? ExceptionType
);

public record ValidationExceptionResponse(
    IDictionary<string, string[]> Errors,
    string CorrelationId,
    string? Message = "One or more validation errors occurred.",
    HttpStatusCode StatusCode = HttpStatusCode.BadRequest
);

public record ValidationExceptionResponseDev(
    IDictionary<string, string[]> Errors,
    string? Details,
    string? StackTrace,
    string? ExceptionType,
    string? Message = "One or more validation errors occurred.",
    HttpStatusCode StatusCode = HttpStatusCode.BadRequest
);

File: client/src/app/models/errorResponse.ts

export interface RTKApiResponse<T> {
    status: number
    data: ApiResponse<T>
}

export interface ApiResponse<T> {
    success: boolean
    status: number,
    data: T
    error: string,
    errors: string[]
}

/* interface ErrorData {
    details: string;
    statusCode: number;
    message: string;
    errors: string[] | null;
    stackTrace: string;
}

export interface ValidationErrorResponse {
    status: number;
    data: string[];
} */
/* Sample data - ValidationErrorResponse
{
    "status": 400,
    "data": [
        "This is the first error",
        "This is the second error"
    ]
}
 */

/* export type ApiErrorResponse = {
    status: number,
    data: ErrorDetail;
}

export type ErrorDetail = {
    error: {
        message: string;
        exceptionType: string;
        errors?: Record<string, string[]>;
    }
}
 */



